app/services/shop/order_cycles_list.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Lists available order cycles for a given customer in a given distributor
+module Shop
+  class OrderCyclesList
```

Comment:
pretty cool what you are lately doing of extracting even more abstractions :clap: :clap: